package/cfg/k3s-cis-1.24-hardened/node.yaml

```diff
@@ -427,17 +398,18 @@ groups:
               path: '{.featureGates.RotateKubeletServerCertificate}'
               set: false
         remediation: |
-          Edit the kubelet service file $kubeletsvc
-          on each worker node and set the below parameter in KUBELET_CERTIFICATE_ARGS variable.
-          --feature-gates=RotateKubeletServerCertificate=true
-          Based on your system, restart the kubelet service. For example:
-          systemctl daemon-reload
-          systemctl restart kubelet.service
+          Not Applicable.
```

Comment:
```suggestion
```
Is there a justification for this check not to be Applicable in K3s ? IMO this should be applicable and changed from Manual to Automated since by default K3s doesn't set this flag, which means that it is enabled.
```
I0724 19:49:02.826935 2798635 check.go:110] -----   Running check 4.2.12   -----
I0724 19:49:04.663575 2798635 check.go:309] Command: "journalctl -u k3s -u k3s-agent | grep -m1 'Running kubelet'"
I0724 19:49:04.663888 2798635 check.go:310] Output:
 "Jul 05 10:52:28 k3s-standalone k3s[2918281]: time=\"2024-07-05T10:52:28Z\" level=info msg=\"Running kubelet --address=0.0.0.0 --allowed-unsafe-sysctls=net.ipv4.ip_forward,net.ipv6.conf.all.forwarding --anonymous-auth=false --authentication-token-webhook=true --authorization-mode=Webhook --cgroup-driver=systemd --client-ca-file=/var/lib/rancher/k3s/agent/client-ca.crt --cloud-provider=external --cluster-dns=10.43.0.10 --cluster-domain=cluster.local --container-runtime-endpoint=unix:///run/k3s/containerd/containerd.sock --containerd=/run/k3s/containerd/containerd.sock --eviction-hard=imagefs.available<5%,nodefs.available<5% --eviction-minimum-reclaim=imagefs.available=10%,nodefs.available=10% --fail-swap-on=false --feature-gates=CloudDualStackNodeIPs=true --healthz-bind-address=127.0.0.1 --hostname-override=k3s-standalone --kubeconfig=/var/lib/rancher/k3s/agent/kubelet.kubeconfig --node-ip=56.56.56.56 --node-labels=rke.cattle.io/machine=48040dc8-62e3-4624-8302-f58cd3a0cb89 --pod-infra-container-image=rancher/mirrored-pause:3.6 --pod-manifest-path=/var/lib/rancher/k3s/agent/pod-manifests --read-only-port=0 --resolv-conf=/run/systemd/resolve/resolv.conf --serialize-image-pulls=false --tls-cert-file=/var/lib/rancher/k3s/agent/serving-kubelet.crt --tls-private-key-file=/var/lib/rancher/k3s/agent/serving-kubelet.key\"\n"
I0724 19:49:04.664177 2798635 check.go:231] Running 2 test_items
I0724 19:49:04.664504 2798635 test.go:153] In flagTestItem.findValue
I0724 19:49:04.664610 2798635 test.go:247] Flag 'RotateKubeletServerCertificate' does not exist
I0724 19:49:04.664639 2798635 check.go:255] Used auditCommand
I0724 19:49:04.664725 2798635 test.go:153] In flagTestItem.findValue
I0724 19:49:04.664783 2798635 test.go:247] Flag 'RotateKubeletServerCertificate' does not exist
I0724 19:49:04.664819 2798635 check.go:255] Used auditCommand
I0724 19:49:04.664838 2798635 check.go:287] Returning from execute on tests: finalOutput &check.testOutput{testResult:true, flagFound:false, actualResult:"Jul 05 10:52:28 k3s-standalone k3s[2918281]: time=\"2024-07-05T10:52:28Z\" level=info msg=\"Running kubelet --address=0.0.0.0 --allowed-unsafe-sysctls=net.ipv4.ip_forward,net.ipv6.conf.all.forwarding --anonymous-auth=false --authentication-token-webhook=true --authorization-mode=Webhook --cgroup-driver=systemd --client-ca-file=/var/lib/rancher/k3s/agent/client-ca.crt --cloud-provider=external --cluster-dns=10.43.0.10 --cluster-domain=cluster.local --container-runtime-endpoint=unix:///run/k3s/containerd/containerd.sock --containerd=/run/k3s/containerd/containerd.sock --eviction-hard=imagefs.available<5%,nodefs.available<5% --eviction-minimum-reclaim=imagefs.available=10%,nodefs.available=10% --fail-swap-on=false --feature-gates=CloudDualStackNodeIPs=true --healthz-bind-address=127.0.0.1 --hostname-override=k3s-standalone --kubeconfig=/var/lib/rancher/k3s/agent/kubelet.kubeconfig --node-ip=56.56.56.56 --node-labels=rke.cattle.io/machine=48040dc8-62e3-4624-8302-f58cd3a0cb89 --pod-infra-container-image=rancher/mirrored-pause:3.6 --pod-manifest-path=/var/lib/rancher/k3s/agent/pod-manifests --read-only-port=0 --resolv-conf=/run/systemd/resolve/resolv.conf --serialize-image-pulls=false --tls-cert-file=/var/lib/rancher/k3s/agent/serving-kubelet.crt --tls-private-key-file=/var/lib/rancher/k3s/agent/serving-kubelet.key\"", ExpectedResult:"'RotateKubeletServerCertificate' is present OR 'RotateKubeletServerCertificate' is not present"}
I0724 19:49:04.665010 2798635 check.go:184] Command: "" TestResult: true State: "PASS"
[INFO] 4 Worker Node Security Configuration
[INFO] 4.2 Kubelet
[PASS] 4.2.12 Verify that the RotateKubeletServerCertificate argument is set to true (Manual)

== Summary node ==
1 checks PASS
0 checks FAIL
0 checks WARN
0 checks INFO
```
More explanations here: https://kubernetes.io/docs/reference/access-authn-authz/kubelet-tls-bootstrapping/#certificate-rotation

Reply:
Addressing this in https://github.com/rancher/security-scan/pull/219